dhall/src/Dhall.hs

```diff
@@ -1341,11 +1374,45 @@ instance (Selector s1, Selector s2, Interpret a1, Interpret a2) => GenericInterp
 
         return (Type {..})
 
-instance Interpret a => GenericInterpret (M1 S s (K1 i a)) where
-    genericAutoWith options = do
-        let Type { extract = extract', ..} = autoWith options
+instance (Selector s, Interpret a) => GenericInterpret (M1 S s (K1 i a)) where
+    genericAutoWith options@InterpretOptions{..} = do
+        let n :: M1 S s (K1 i a) r
+            n = undefined
```

Comment:
I'm a little concerned about all these `undefined`s. Would `Proxy` be a safer alternative?

Reply:
@rprije: The `undefined`s are necessary.  It's a limitation of the GHC generics API, since it doesn't support a `Proxy`-based `selName`.